shoper.py

```diff
@@ -0,0 +1,135 @@
+import os
+import pandas as pd
+from operator import itemgetter
+from re import search
+
+
+def deactivate_products(database, dc_price, replacement_price):
+    """
+    Deactivates products set with exact price - e.g. 107.99
+    Products are kept because some of them have been withdrawn,
+    yet they might come back
+    
+    csv_file: .csv file to work with
+    dc_price: exact price that triggers deactivation of product
+    column_types: column types ensuring proper encoding of file
+    """
+    print("\nDeactivated following products:")
+    
+    for row in range(database.shape[0]):
+        
+        if database['price'][row]:
+            database.at[row,'price'] = replacement_price
+            database.at[row, 'active'] = 0
+            print(database['name'][row],"at code", database['product_code'][row])
+            
+        if pd.isnull(database.loc[row, 'price']) == True:
+            database.at[row, 'price'] = replacement_price
+            database.at[row, 'active'] = 0
+            print(database['name'][row],"at code", database['product_code'][row])
+    
+
+def repair_dateprices(database, months):
+    """
+    Repairs date prices (10.99 [meant] -> oct.99 [edited] -> 10.99)
```

Comment:
Ten komentarz nie ma sensu w j. angielskim. Naprawia daty cen? Nie wiem czy to jest 10.99 zł czy 10.99 (październik '99). Docstring do naprawy! Strzałki mylą o co chodzi i co jest końcowym produktem, jeśli meant jest 1, a strzałki lecą dalej.